src/sentry/monitors/endpoints/organization_monitor_details.py

```diff
@@ -114,6 +128,27 @@ def put(self, request: Request, organization, project, monitor) -> Response:
             params["status"] = result["status"]
         if "config" in result:
             params["config"] = result["config"]
+
+            # update timeouts + expected next check-in, as appropriate
+            checkin_margin = result["config"].get("checkin_margin")
+            if checkin_margin != existing_margin:
+                for monitor_environment in MonitorEnvironment.objects.filter(monitor_id=monitor.id):
+                    monitor_environment.next_checkin_latest = (
+                        monitor_environment.next_checkin
+                        + timedelta(minutes=int(checkin_margin or 1))
+                    )
```

Comment:
Let's find a way to reuse this code

https://github.com/getsentry/sentry/blob/09fce5932d6a24051219c6dd9a9482660d3ba310/src/sentry/monitors/models.py#L249-L259

maybe we can extract it into a helper function. I don't want to duplicate this magic default number all over the place.

Reply:
i'll see what I can do but the `F()` query is a bit hard to extract